opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp

```diff
@@ -213,10 +213,22 @@ class EclMaterialLawManager
         return *materialLawParams_[elemIdx];
     }
 
-    std::shared_ptr<MaterialLawParams>& materialLawParamsPointerReferenceHack(unsigned elemIdx)
+    MaterialLawParams& materialLawParamsWellConnection(unsigned elemIdx)
     {
         assert(0 <= elemIdx && elemIdx <  materialLawParams_.size());
-        return materialLawParams_[elemIdx];
+        return *materialLawParams_[elemIdx];
+    }
+
+    const MaterialLawParams& materialLawParamsWellConnection(unsigned int connectionIdx) const
```

Comment:
> If the completions can change during the simulation

A completion can change some state during the simulation - but it can not move; that will be a *new* completion.